### PR TITLE
[dashboard] improve branches page

### LINKED ISSUE #5585: [projects] Allow navigation on branches and commits
The branches view as well as the preview, shows branches, but don't allow going to the context (git repo).

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description
Some improvements on the branches page
 - allow navigating to the repository
 - allow navigation to the branches

Screenshot 2021-09-28 at 09 56 17

## Related Issue(s)
Fixes #5585

## How to test
Create a project and see the changes on the branches page.

## Release Notes
```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **svenefftinge** after the PR has been reviewed.

Associated issue: *#5585*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by gtsiolis** [comment]
Looking at this now! 👀

File: components/dashboard/src/projects/Project.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/Project.tsx:174]: **thought**: I'd love to also link to each commit from the commit hash in the column as described in the relevant issue in https://github.com/gitpod-io/gitpod/issues/5585. However, we can skip if for this PR and open a follow-up issue for this. 🏓

File: components/dashboard/src/projects/Project.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/Project.tsx:132]: **issue(non-blocking)**: Replacing the project name with the repository URL could make long paths not scale well (see [example](https://gitlab.com/gitlab-com/marketing/community-relations/contributor-program/general/)). However, this a) is probably an edge case for now and b) could be resolved once we have an overview page where we list full details for a project.

Screenshot 2021-09-29 at 11 19 59 AM

Comment (gtsiolis) [inline components/dashboard/src/projects/Project.tsx:132]: **issue(non-blocking)**: The blue color seems slightly distracting up there but helpful. Also, this could be resolved once we have a better overview page.

File: components/dashboard/src/projects/Project.tsx
Comment (gtsiolis) [inline components/dashboard/src/projects/Project.tsx:174]: **issue**: This is missing a hover state! 🙀

We could go with something subtle for now like the following. What do you think?

```suggestion
                                <a href={branch.url}><div className="text-base text-gray-600 hover:text-gray-800 dark:text-gray-50 dark:hover:text-gray-200 font-medium mb-1">
```

**Review by gtsiolis — APPROVED** [review]
Thanks @svenefftinge! Left some minor comments, most of them non-blocking. 🏓 

Approving to unblock merging but holding in case you think any of the issues are worth addressing in this PR. 🏓 

/hold

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 6780cce88642a39d74e2ec87dc4115935f7506c1</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5587#pullrequestreview-766323571" title="Approved">gtsiolis</a>*

Associated issue: *#5585*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 1a5b00fbf16976a350ddaac9b93b9ff51b8e92a6</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5587#issuecomment-930092716" title="LGTM">gtsiolis</a>*

Associated issue: *#5585*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/dashboard/src/projects/Project.tsx
Comment (gtsiolis) [reply components/dashboard/src/projects/Project.tsx]: **fyI**: Added follow up issue in https://github.com/gitpod-io/gitpod/issues/5923.

**Pull request merged**
